Sources/SPMBuildCore/PluginInvocation.swift

```diff
@@ -383,14 +382,16 @@ extension PackageGraph {
                 class PluginDelegate: PluginInvocationDelegate {
                     let delegateQueue: DispatchQueue
                     let toolPaths: [AbsolutePath]
+                    let fileSystem: FileSystem
                     var outputData = Data()
                     var diagnostics = [Basics.Diagnostic]()
                     var buildCommands = [BuildToolPluginInvocationResult.BuildCommand]()
                     var prebuildCommands = [BuildToolPluginInvocationResult.PrebuildCommand]()
                     
-                    init(delegateQueue: DispatchQueue, toolPaths: [AbsolutePath]) {
+                    init(delegateQueue: DispatchQueue, toolPaths: [AbsolutePath], fileSystem: FileSystem) {
```

Comment:
nit: make `fileSystem` the first argument (we usually so). otherwise looks great!

Reply:
Is there a reason why it's made to be the first argument in general?